src/model/model.rs

```diff
@@ -257,7 +260,12 @@ pub enum AddressPattern {
     IpAddr { addr: IpAddr, prefix: u8 },
     Domain {
         #[serde(with = "serde_regex")]
-        pattern: Regex,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        pattern: Option<Regex>,
+        #[serde(default)]
+        #[serde(skip_serializing_if = "Option::is_none")]
+        wildcard: Option<String>,
```

Comment:
It seems to be that `pattern` and `wildcard` should not be `Some` (or `None`) simultaneously.
Such invariants should be represented by sum type (= `enum` in rust).

Reply:
More specifically, for example,

```
// pseudo code
enum AddressPattern = IpAddr | Domain(DomainPattern)
enum DomainPattern = Regex | Wildcard
```

Reply:
done with https://github.com/Idein/gatekeeper/pull/44/commits/f52e878a8bb69d42af32c5bed84917affd7bd42c